### PR TITLE
Create generate.py

### LINKED ISSUE #38: Add features to #5 and #16 (stochastic and bit parity tests)
Add the following features to the probabilistic (#5) and stochastic (#16) tests:

1. Make it clear in the plot whether we are evaluating on test or training set
2. Make it such that n (sequence length) and N (sample size) are different by introducing sampling with replacement.
3. Plot distributions of P(0) (probability associated with 0) and P(1) (probability associated with 1), and check if P(0) and P(1) sum up to 1. 
4. Think about the saving epochs issues
5. Add to the workflow test of whether it is fine-tuning (#37) before evaluation

More context: https://docs.google.com/document/d/1rZFhzmBDRbMvJRqs1Z0sv4R-7iFoC-fqDrY0NT4JRiA/edit?usp=sharing

### PR DESCRIPTION
**Pull request by alessandramaranca**
Generate bit_squences finetuning tests (bit parity and probabilistic tests). This generates a dataset of binary sequences of a given length (`--bit_length`) and total size (`--N`), then splits it into train/test (`--test_size`). You choose labeling mode with `--bit_parity True` to assign outputs by parity (optionally flipping labels with probability `--p`), or `--bit_parity False` to sample outputs as independent Bernoulli draws with probability `--p`. It validates inputs (`2^bit_length ≤ N`, `0 ≤ p ≤ 1`, etc.), and writes JSON files (`train.json` and `test.json`) to the specified output directory.

Closes #38 

# Description

Made it such that sequence length and sample size are different by introducing sampling with replacement. Made the code more generalizable, added assert statements, integrated with our current workflow (no repetitions). 

# Testing Instructions

This integrates with our current finetuning and evaluation workflow (just finetune on file train.json and evaluate/test on file test.json).

### CODE REVIEW

File: tests/bit_sequences/generate.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+"""
+Generate binary sequence datasets with either parity-based or probabilistic labeling to any outdir.
+
+Usage examples:
+  # Parity dataset (no noise), 2^5=32 unique, N=1000 samples, test_size=200
+  python generate.py --bit_length 5 --N 1000 --test_size 200 --bit_parity True --test_size 1000
```
Comment (niznik-dev) [inline tests/bit_sequences/generate.py:7]: Agreed that there's a duplicate `test_size` (thanks Copilot!)

**Review by niznik-dev — CHANGES_REQUESTED** [review]
I think if you fix the typo and add the README.md I'm good with this being merged

**Comment by msalganik** [comment]
Thanks @niznik-dev for the quick review.  And thanks copilot!

**Comment by alessandramaranca** [comment]
Typo fixed and readme added!
